softioc/device.py

```diff
@@ -236,20 +266,37 @@ class GenericDevice(Base):
 _In = ProcessDeviceSupportIn
 _Out = ProcessDeviceSupportOut
 
-def _Device_In(type, **kargs):
-    return _Device(_In,  type, **kargs)
+def _Device_In(record_type, value_to_epics, **kargs):
+    return _Device(_In,  record_type, value_to_epics, **kargs)
+
+def _Device_Out(record_type, value_to_epics, convert=True, mlst=True, **kargs):
+    return _Device(_Out,
+                   record_type,
+                   value_to_epics,
+                   convert=convert,
+                   mlst=mlst,
+                   **kargs)
 
-def _Device_Out(type, convert=True, mlst=True):
-    return _Device(_Out, type, convert=convert, mlst=mlst, default=None)
 
-longin = _Device_In('longin')
-longout = _Device_Out('longout')
-bi = _Device_In('bi', convert=False)
-bo = _Device_Out('bo', convert=False)
-stringin = _Device_In('stringin', mlst=False, default='')
-stringout = _Device_Out('stringout', mlst=False)
-mbbi = _Device_In('mbbi', convert=False)
-mbbo = _Device_Out('mbbo', convert=False)
+def epics_string(value):
+    """Trim a string to EPICS 40 (39 with null byte) character limit"""
+    if value is None:
+        raise ValueError("\"None\" not a valid value")
+
```

Comment:
I took the indication that `None` was never a valid value quite literally and made all records reject it completely.

I think I prefer to leave this in, as it's the sort of error that may creep in during a running IOC and then they end up with some output string reporting "None", rather than an expected message. There's also then confusion if a user ever validly wants to write "None" as their value.
Then again, raising the exception is technically undocumented behaviour and so the IOC may well not catch it, and thus stop running altogether. What guidelines does PythonSoftIOC follow regarding trying to keep running in all cases?

Reply:
After discussion I'll remove this special case check - there's so many other special cases that catching just this one isn't particularly worthwhile.